spec/derivative_rodeo/storage_adapters/sqs_adapter_spec.rb

```diff
@@ -0,0 +1,56 @@
+# frozen_string_literal: true
+
+RSpec.describe DerivativeRodeo::StorageAdapters::SqsAdapter do
+  let(:file_path) { File.expand_path(File.join(FIXTURE_PATH, 'files', 'ocr_color.tiff')) }
+  let(:short_path) { file_path.split('/')[-2..-1].join('/') }
+  let(:args) { "sqs://eu-west-1.amazonaws.com/55555555/fake-queue/#{short_path}?template=s3://adventist-preprocess.s3.us-west-1.amazonaws.com/preprocess/#{short_path}" }
+  let(:client) { AwsSqsFauxClient.new }
+
+  subject { described_class.new(args) }
+
+  before do
+    DerivativeRodeo.config do |config|
+      config.aws_sqs_queue = 'fake-queue'
+      config.aws_sqs_account_id = '55555555'
+      config.aws_sqs_access_key_id = "FAKEFAKEFAKE"
+      config.aws_sqs_secret_access_key = "FAKEFAKEFAKEFAKER"
+    end
+  end
+
+  it "creates a properly formatted uri from a file path with 2 path parts as default" do
+    expect(described_class.create_uri(path: file_path)).to eq("sqs://us-east-1.amazonaws.com/55555555/fake-queue/ocr_color.tiff")
+  end
+
+  it "creates a properly formatted uri from a file path with all parts" do
+    expect(described_class.create_uri(path: file_path, parts: :all)).to eq("sqs://us-east-1.amazonaws.com/55555555/fake-queue/#{file_path}")
```

Comment:
Are we expecting `/fake-queue//Users/jeremyf/path/to/file` note the double `\\` between the adapter prefix and the file.

Reply:
nope, thats a bug